.github/workflows/integration-tests.yml

```diff
@@ -5,81 +5,32 @@ on:
   workflow_dispatch:
 
 jobs:
-  run-tests:
-    runs-on: ubuntu-latest
-    timeout-minutes: 45
-    strategy:
-      matrix:
-        # These are all the Python versions that we support.
-        python-version: ['3.7', '3.8', '3.9', '3.10']
-    
-    name: Run Fast Integration Tests with Python Version ${{ matrix.python-version }}
+  run-tests-aqueduct-engine-and-db:
+    runs-on: ubuntu-latest-8-cores
+    timeout-minutes: 20
+
+    name: Run all Integration Tests with Aqueduct Engine and DB
     steps:
       - uses: actions/checkout@v2
 
-      - name: Create the logs directory
-        run: mkdir -p logs
-
-      - name: Set filename for the server's output logs
-        run: echo "SERVER_LOGS_FILE=logs/server_logs_${{ matrix.python-version }}" >> $GITHUB_ENV
-
-      - name: Set up Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python-version }}
-
-      - name: Set up GOPATH variable
-        run: echo "GOPATH=$(echo $HOME)" >> $GITHUB_ENV
-
-      - name: Set up Go
-        uses: actions/setup-go@v2
-        with:
-          go-version: 1.19.1
-      
-      - name: Install aqueduct-ml
-        run: python3 -m pip install aqueduct-ml
-
-      - name: Install compatible SQLAlchemy
-        run: pip3 install SQLAlchemy==1.4.46
-
-      - name: Start the server
-        run: (aqueduct start --verbose > $SERVER_LOGS_FILE 2>&1 &)
-
-      - name: Wait for server
-        timeout-minutes: 2
-        run: while ! echo exit | nc localhost 8080; do sleep 1; done
-
-      # Grabs the pid of the process bound to port 8080 and kills it.
-      - name: Kill the server
-        run: kill -9 $(lsof -nP -iTCP -sTCP:LISTEN | grep 8080 | awk '{print $2}')
-
-      # install_local.py requires ~/.aqueduct to exist.
-      - name: Update aqueduct with latest code
-        run: python3 scripts/install_local.py -g -s -e
-
-      - name: Start the server again
-        run: (aqueduct start --disable-usage-stats --verbose > $SERVER_LOGS_FILE 2>&1 &)
-
-      - name: Install packages needed for testing
-        run: python3 -m pip install nltk matplotlib pytest-xdist
-
-      - name: Wait for server again
+      - uses: ./.github/actions/setup-server
         timeout-minutes: 5
-        run: while ! echo exit | nc localhost 8080; do sleep 1; done
-
-      - name: Fetch the API key
-        run: echo "API_KEY=$(aqueduct apikey)" >> $GITHUB_ENV
 
       - name: Set up the SDK Integration Tests
         working-directory: integration_tests/sdk
         run: |
-          echo "apikey: $(aqueduct apikey)" >> test-config.yml
-          echo "address: localhost:8080" >> test-config.yml
+          echo "apikey: $(aqueduct apikey)" >> test-credentials.yml
```

Comment:
so for non aq data / compute testings they will use another action?

Reply:
not another action, another job in the same action.

Reply:
is the job added? I only find this job which runs the default settings, that's why I'm a bit confused

Reply:
it's not added because the plumbing isn't quite ready yet (eg. we still need to setup the data integrations on the fly)

Reply:
sounds cool! Just making sure I'm understanding the context